src/api/uploads/MultiputUpload.js

```diff
@@ -0,0 +1,367 @@
+/**
+ * @flow
+ * @file Multiput upload
+ * @author Box
+ */
+
+import Base from '../Base';
+import { getFileLastModifiedAsISONoMSIfPossible } from '../../util/uploads';
+import createWorker from '../../util/uploadsSHA1Worker';
+import MultiputPart, { PART_STATE_UPLOADED, PART_STATE_DIGEST_READY } from './MultiputPart';
+import type { StringAnyMap } from '../../flowTypes';
+
+const DEFAULT_MULTIPUT_CONFIG = {
+    console: false,
+    digestReadahead: 5,
+    initialRetryDelayMs: 5000,
+    maxRetryDelayMs: 60000,
+    parallelism: 5,
+    partSize: null,
+    requestTimeout: 120000,
+    retries: 5
+};
+
+class MultiputUpload extends Base {
+    clientId: ?string;
+    commitRetryCount: number;
+    config: Object;
+    createSessionNumRetriesPerformed: number;
+    destinationFile: ?string;
+    destinationFolder: ?string;
+    ended: boolean;
+    file: File;
+    fileSha1: ?string;
+    firstUnuploadedPartIndex: number;
+    initialFileLastModified: ?string;
+    initialFileSize: number;
+    onCompleted: ?Function;
+    onFailure: ?Function;
+    onProgress: ?Function;
+    partSize: number;
+    parts: Array<MultiputPart>;
+    partsDigestComputing: number;
+    partsDigestReady: number;
+    partsNotStarted: number;
+    partsUploaded: number;
+    partsUploading: number;
+    sessionEndpoints: Object;
+    totalUploadedBytes: number;
+    worker: Worker;
+    sessionId: ?string;
+
+    /**
+     * [constructor]
+     * 
+     * @private
+     * @param {object} options
+     * @param {File} file
+     * @param {string} createSessionUrl
+     * @param {string} [destinationFolder] - Untyped folder id (e.g. no "d_" prefix)
```

Comment:
Is there any notion of "typed" ids in the public API or in the elements codebase? If not, then we should modify these comments